include/merlin_hashtable.cuh

```diff
@@ -579,6 +579,230 @@ class HashTable {
     CudaCheckError();
   }
 
+  /**
+   * @brief Searches the hash table for the specified keys.
+   * When a key is missing, the value in @p values and @p metas will be
+   * inserted.
+   *
+   * @param n The number of key-value-meta tuples to search or insert.
+   * @param keys The keys to search on GPU-accessible memory with shape (n).
+   * @param values The values to search on GPU-accessible memory with
+   * shape (n, DIM).
+   * @param metas The metas to search on GPU-accessible memory with shape (n).
+   * @parblock
+   * If @p metas is `nullptr`, the meta for each key will not be returned.
+   * @endparblock
+   * @param stream The CUDA stream that is used to execute the operation.
+   *
+   */
+  void find_or_insert(const size_type n, const key_type* keys,  // (n)
```

Comment:
Hi @WonderingWJ , please help confirm the API define is matching your requirement, thank you~